### ISSUE TITLE
No cover  on books imported with kobocloud after new upgrade

### ISSUE THREAD

**Issue by sidecar12** [fsantini/KoboCloud#136]
Strange thing appare in my kobo after firmware upgrade 4.32.19501 .. if i charge the book with kobocloud (dropbox) the book is readble but it dont have a cover  in the homepage instead of that if i charge the same book with pc throught usb all works fine.
Note: the book have inside the correct cover.

Any idea or solution

best regard enrico

**Comment by Krzycho666** [comment]
The same issue on Kobo Libra H20 after downloading 4.32.19501

**Comment by duiffie** [comment]
Same here on a Kobo Forma running 4.32.19501

**Comment by williamutbult** [comment]
Same here. Kobo Libra H2O after updating to 4.32.19501(0e55e3f620, 4/14/22).

**Comment by anilkarip** [comment]
Same issue with Clara HD at firmware 4.32.19501

**Comment by groentebroer** [comment]
Anyone knows this makes a difference between a "kepub" or "epub"?
Anyone has some more information on what might have changed in the Firmware? Or that it is (hidden) folder related or something?

**Comment by groentebroer** [comment]
I did a small test.
I uploaded the same book, as KEPUB (file extension: .kepub.epub) and as EPUB (file extension .epub) through KoboCloud.

It displays the cover correctly in the overview for the KEPUB format.
It does not display the cover in the overview for the EPUB format. So a workaround is at least saving the EPUB to KEPUB (and properly name the file to work with KoboCloud correctly (e.g. .kepub.epub)).

**Comment by EnzoPoint** [comment]
Same issue for me 😔, can you please try to fixe this ?

**Comment by creekmoreworld** [comment]
Yeah, same.  I spent a long time trying to nail it down.  I think it has to be this.  Kobo Libra H2O.
Thanks for your work!

**Comment by Abushawish** [comment]
Same issue here with my Clara

**Comment by micdabe** [comment]
Same here with Kobo Libra 2
Solved with kepub conversion

**Comment by dirtbaghippy** [comment]
Also having this issue w/libra h2o

**Comment by maetthew** [comment]
Experiencing same issue with Libra H2O.

**Comment by salama288** [comment]
Same here with kobo aura one 4.34

**Comment by vnoel** [comment]
Same for me. Aura H2O.

**Comment by DylanDOT** [comment]
Same issue here on Kobo Clara 2e

**Comment by Bastet91** [comment]
Same problem with Kobo Clara HD. Is there any solution?

**Comment by isaacgr** [comment]
Is this repo still active? Having this problem as well but doesn't look like a lot of activity. 

Love the tool btw, but just wondering if there are more up to date alternatives?

**Comment by fsantini** [comment]
Hi, I'm still keeping an eye on it, but at the moment it's on a "works for me" state and I don't have the resources to fix the problems. I also have a quite old Kobo and can't reproduce most of the problems. Thanks for understanding.

**Comment by groentebroer** [comment]
@isaacgr see the above workaround I posted.
I use Calibre to create the kepub format, and give it the extension '.epub', then the covers are displayed fine.

**Comment by divx118** [comment]
The problem is the way how importing the new files work. The bind mount to a virtual sd card gives me double issues. no cover and double the books. For each file there are 2 books added to the library. This will happen after a connect reconnect with USB when it imports the book correctly with cover. 
BTW I am using google cloud and adobe digital edition books with DRM on a libra 2.
I am poking around telnet to see if there is any other way to update the library. For now I removed the bindmounts in the code. https://github.com/fsantini/KoboCloud/blob/master/src/usr/local/kobocloud/get.sh#L85 and https://github.com/fsantini/KoboCloud/blob/cf7b0bf9d177bfb1a389f719770547f30da5b550/src/usr/local/kobocloud/udev_mount.sh#L20

**Comment by divx118** [comment]
Ok after much poking around nothing simple popped up to rescan the library. So I installed https://github.com/shermp/NickelDBus and just put the following command in get.sh and udev_mount.sh. Which works very well. 
`qndb -t 30000 -s pfmDoneProcessing -m pfmRescanBooksFull`

**Comment by fsantini** [comment]
Qndb is now used in the latest version of the original KoboCloud. I believe that the cover issue is mitigated by converting the books to kepubs using kepubify. However, I noticed it doesn't always work.

**Comment by thomas-negrault** [comment]
I confirm that it does not always work, I tested 2 ebooks with the online version of Kebubify and I don't have the cover on my Kobo.

**Comment by svenbox** [comment]
Just tried KoboCloud and this seems to be an issue still - no cover image. Can anyone else confirm?

**Comment by MoonshineSG** [comment]
any permanent solution for this ? I just added kobocloud to my Libra2 and no cover on any. of the GDrive books...

**Comment by svenbox** [comment]
> any permanent solution for this ? I just added kobocloud to my Libra2 and no cover on any. of the GDrive books...

Use the .kepub format and the problems are gone!

**Comment by MoonshineSG** [comment]
> > any permanent solution for this ? I just added kobocloud to my Libra2 and no cover on any. of the GDrive books...
> 
> 
> 
> Use the .kepub format and the problems are gone! 

nope. didn't work for me...
I gave up and switched to calibre... more complex setup but seams to work well... 🤞🏻

**Comment by anilkarip** [comment]
Covers not working that's it :) I installed Dropbox extension. Similar but working solution.

**Comment by thomas-negrault** [comment]
I managed to get covers converting to kebub. Here is what I've done (on Linux/Ubuntu):
- Download kebubify from https://pgaskin.net/kepubify/dl/
- Run `chmod +x {file}`
- Run `Kepubify\ Linux\ 64bit /path/with/your/epubs`

This will convert your epub to kebub in a `._converted` folder, move these files to your cloud provider and synchronize on your tablet

Full kepubify instructions here: https://pgaskin.net/kepubify/docs/#getting-started

**Comment by 5kxh2dxqmd-afk** [comment]
Kepubify works for my EPUBs, but I want to get PDF covers as well. Is there a way to do this?
